diff --git a/include/oneapi/dnnl/dnnl.hpp b/include/oneapi/dnnl/dnnl.hpp
index 9114c0a5f..28907d98f 100644
--- a/include/oneapi/dnnl/dnnl.hpp
+++ b/include/oneapi/dnnl/dnnl.hpp
@@ -4686,6 +4686,8 @@ struct convolution_forward : public primitive {
     struct desc {
         dnnl_convolution_desc_t data;
 
+        desc() {}
+
         /// Constructs a descriptor for a convolution forward propagation
         /// primitive with bias.
         ///
@@ -5433,6 +5435,8 @@ struct deconvolution_forward : public primitive {
     struct desc {
         dnnl_deconvolution_desc_t data;
 
+        desc() {}
+
         /// Constructs a descriptor for a deconvolution forward propagation
         /// primitive with bias.
         ///
@@ -6390,6 +6394,8 @@ struct pooling_forward : public primitive {
     struct desc {
         dnnl_pooling_desc_t data;
 
+        desc() {}
+
         /// Constructs a descriptor for pooling forward propagation primitive.
         ///
         /// Arrays @p strides, @p kernel, @p padding_l, and @p padding_r
@@ -8241,6 +8247,8 @@ struct inner_product_forward : public primitive {
     struct desc {
         dnnl_inner_product_desc_t data;
 
+        desc() {}
+
         /// Constructs a descriptor for an inner product forward propagation
         /// primitive with bias.
         ///
@@ -11972,6 +11980,8 @@ struct matmul : public primitive {
     struct desc {
         dnnl_matmul_desc_t data;
 
+        desc() {}
+
         /// Constructs a descriptor for a matmul primitive.
         ///
         /// @param src_desc Memory descriptor for source (matrix A).
